contentcuration/contentcuration/frontend/shared/data/resources.js

```diff
@@ -189,9 +193,19 @@ class IndexedDBResource {
     copyProperties(this, options);
     // By default these resources do not sync changes to the backend.
     this.syncable = syncable;
+    // An object for listening to specific change events on this resource in order to
+    // allow for side effects from changes - should be a map of change type to handler function.
+    this.listeners = listeners;
   }
 
   get table() {
+    if (process.env.NODE_ENV !== 'production' && !db[this.tableName]) {
+      /* eslint-disable no-console */
+      console.error(
+        `Tried to access table ${this.tableName} but it does not exist. Either requires a migration or clearing indexedDB`
```

Comment:
We'll probably want to have a way to handle this in production at some point

Reply:
Yeah, we will need to add explicit versions to the `IndexedDBResources` if we need to add new ones.

I added this warning because I had switched back to this branch and so didn't have the Task table, and it just silently failed.